src/pages/[platform]/build-a-backend/storage/download/index.mdx

```diff
@@ -167,34 +207,74 @@ try {
   console.log('Error : ', error);
 }
 ```
+</Block>
+</BlockSwitcher>
 
 ### Track the progress of a download task
 
 To track the progress of your download, you can use `onProgress`:
 
+<BlockSwitcher>
+<Block name="With Path (Above v6.x.x)">
+```javascript
+import { downloadData } from 'aws-amplify/storage';
+
+// Download a file from s3 bucket
+const { body, eTag } = await downloadData(
+  {
+    path: "/album/2024/1.jpg",
+    options: {
+      onProgress: (progress) {
+        console.log(`Download progress: ${(progress.transferredBytes/progress.totalBytes) * 100}%`);
+      }
+    }
+  }
+).result;
+```
+</Block>
+
+<Block name="With Key (Deprecated)">
 ```javascript
 import { downloadData } from 'aws-amplify/storage';
 
 // Download a file from s3 bucket
 const { body, eTag } = await downloadData(
   {
-    key,
-    data: file,
+    key: "abc.txt",
     options: {
       onProgress: (progress) {
-        console.log(`Downloaded: ${progress.transferredBytes}/${progress.totalBytes}`);
+        console.log(`Download progress: ${(progress.transferredBytes/progress.totalBytes) * 100}%`);
       }
     }
   }
 ).result;
 ```
+</Block>
+</BlockSwitcher>
 
 ### Cancel a download task
 
+<BlockSwitcher>
+<Block name="With Path (Above v6.x.x)">
 ```javascript
 import { downloadData, isCancelError } from 'aws-amplify/storage';
 
-const downloadTask = downloadData({ key, data: file });
+const downloadTask = downloadData({ path: "/album/2024/1.jpg", data: file });
```

Comment:
nit: ditto can break multiple properties object into multilines to avoid horizontal scrolling or softwraps